### PR TITLE
Add bufr2ioda ctest for marine obs profiles

### PR DESCRIPTION
**Pull request by ShastriPaturi**
This PR adds bufr2ioda ctests for marine obs profile that includes both temp and salinity
Daily Tanks:
1. dbuoy: drifting buoy (b001/xx002: old tank)
2. mbuouy: moored buoy (b001/xx103: new tank)
Monthly Tanks:
1. tesac: Temperature and Salinity data (b031/xx002)
2. trkob: Temperature and Salinity data (b031/xx003)

### CODE REVIEW

**Comment by ShastriPaturi** [comment]
some unit tests are failing. I am not sure why. I have only tested for the ctests I have added.

File: test/CMakeLists.txt [hunk 1]
```diff
@@ -1,7 +1,7 @@
 ##### get test data from EMC FTP server
 # set URL and hash
 set(URL "https://ftp.emc.ncep.noaa.gov/static_files/public/GDASApp")
-set(SHA "2f85f9469ee0788b3765a667fa6e9cd52b599dfb986b7dc773aaf4bcdaf05b4b")
+set(SHA "43c9e0990ea663e6da1c21121391c10731f53f31838dae112e0e63e5c38f6af7")
```
Comment (CoryMartin-NOAA) [inline test/CMakeLists.txt]: Is this the right hash? I think this might be why the unit test is failing.

Comment (ShastriPaturi) [reply test/CMakeLists.txt]: It is the hash you shared to get the subsampled files

Comment (CoryMartin-NOAA) [reply test/CMakeLists.txt]: Yes but that was over 6 weeks ago. Russ updated that file 28 days ago in https://github.com/NOAA-EMC/GDASApp/pull/139

**Review by guillaumevernieres — CHANGES_REQUESTED** [review]
1. I don't think a license header is needed for your `yaml` files. @RussTreadon-NOAA ? @CoryMartin-NOAA ? And if it is, the year is wrong.
2. You're missing ctests for `mbuoyb`, `tesac` and`trkob`. Remove the unused yaml's or add ctests, whatever makes more sense.

**Comment by CoryMartin-NOAA** [comment]
> 1. I don't think a license header is needed for your `yaml` files. @RussTreadon-NOAA ? @CoryMartin-NOAA ? And if it is, the year is wrong.
> 
>     2. You're missing ctests for `mbuoyb`, `tesac` and`trkob`. Remove the unused yaml's or add ctests, whatever makes more sense.

Yeah we really shouldn't be putting copyright statements in things. It's pervasive in the JEDI repos, but all NOAA stuff is taxpayer funded and needs to be publicly available anyways.

File: test/soca/CMakeLists.txt [hunk 2]
```diff
@@ -29,6 +29,25 @@ endforeach(FILENAME)
 # tests below require the full build
 if(BUILD_GDASBUNDLE)
 
+  # link input file from iodaconv to test directory
+  #file(CREATE_LINK ${PROJECT_SOURCE_DIR}/iodaconv/test/testinput/dbuoy_20180415_bufr ${PROJECT_BINARY_DIR}/test/testdata/dbuoy_20180415_bufr SYMBOLIC)
```
Comment (CoryMartin-NOAA) [inline test/soca/CMakeLists.txt]: remove commented out line

**Comment by ShastriPaturi** [comment]
@CoryMartin-NOAA and @guillaumevernieres the requested changes are addressed.
@CoryMartin-NOAA can you remove dbuoyb.20180415_subsampled and bathy.201804.dcom_subsampled from the tarball? I will revisit them later as the subsampled files are failing. The original files work fine but too large (>5MB) to be included.

**Comment by CoryMartin-NOAA** [comment]
@ShastriPaturi if you plan to revisit it later, I would prefer to keep the files there since that means I don't have to update with yet another hash and have another version of the tar ball archived on rzdm.

File: test/soca/CMakeLists.txt (hunk 2 above)
Comment (CoryMartin-NOAA) [reply test/soca/CMakeLists.txt]: This change still needs addressed @ShastriPaturi

**Comment by ShastriPaturi** [comment]
> @ShastriPaturi if you plan to revisit it later, I would prefer to keep the files there since that means I don't have to update with yet another hash and have another version of the tar ball archived on rzdm.

Fair enough @CoryMartin-NOAA

File: test/soca/CMakeLists.txt (hunk 2 above)
Comment (ShastriPaturi) [reply test/soca/CMakeLists.txt]: Done

**Review by CoryMartin-NOAA — APPROVED** [review]
Thanks @ShastriPaturi I'm happy whenever @guillaumevernieres is happy we can merge

**Review by guillaumevernieres — APPROVED** [review]
Exiting stuff! Thanks @ShastriPaturi .

**Pull request merged**